### PR TITLE
Set Up Testing for project

### LINKED ISSUE #18: Set up testing for our ecommerce frontend
### Description

We need to set up a robust testing environment for our TypeScript + Vite + React project. The testing setup should include unit tests, integration tests, and end-to-end tests to ensure the reliability and stability of our codebase.

### Tasks

- [x] Install Testing Libraries and Tools
- [x] Create a `vitest.config.js` file with the necessary configuration for TypeScript and React.
- [x] Integrate React Testing Library with vitest
- [x] Create a setup file for React Testing Library
- [x] Write sample unit tests for a simple React component.

### Acceptance Criteria

1. Jest is installed and configured with TypeScript and Vite.
2.  React Testing Library is integrated with Jest.
3. Cypress is installed and configured.
4. Sample unit tests, integration tests, and end-to-end tests are written and pass successfully.
5. Documentation is updated with instructions on how to run the tests.

### PR DESCRIPTION
**Pull request by niyobern**
### Description:

This PR introduces testing setup for our Vite + React + TypeScript project. The goal is to ensure that our components and business logic function as expected, and to catch any potential issues before they reach production.

### Changes:

- Installed [vitest](https://vitejs.dev/) and `@testing-library/react` as dev dependencies.
- Created a test file [App.test.tsx](https://github.com/atlp-rwanda/dynamites-ecomm-fe/src/__test__/App.test.ts) for the [App](https://github.com/atlp-rwanda/dynamites-ecomm-fe/src/App.ts) component.
- Wrote a test case to check if the [App](https://github.com/atlp-rwanda/dynamites-ecomm-fe/src/App.ts) component renders the expected content.

### Test:

The test checks if the [App]() component renders a paragraph with the text 'Welcome to Dynamites E-commerce'. This is done by rendering the [App]() component and then using [screen.getByRole('paragraph')]() to find the paragraph element. The [toHaveTextContent]() assertion is then used to check if the paragraph contains the expected text.

### Impact:

This change doesn't affect the existing functionality of the application but provides a foundation for writing more comprehensive tests in the future. This will help us ensure the quality and reliability of our application as we continue to develop and add new features.

### How to Test:

After pulling the changes, run 
```bash
npm run test
```
to test the [App](/https://github.com/atlp-rwanda/dynamites-ecomm-fe/src/App.ts) component

### CODE REVIEW

**Review by 13XAVI — APPROVED** [review]

File: vite.config.ts [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
+/// <reference types="vite/client" />
+
 import { defineConfig } from 'vite';
 import react from '@vitejs/plugin-react';
 import path from 'path';
 
 // vite.config.ts
 export default defineConfig({
   plugins: [react()],
-  server: {
```
Comment (jkarenzi) [inline vite.config.ts]: why did you remove this?

**Review by jkarenzi — APPROVED** [review]

File: vite.config.ts (hunk 1 above)
Comment (niyobern) [reply vite.config.ts]: rebase error

**Review by jkarenzi — APPROVED** [review]

File: vite.config.ts (hunk 1 above)
Comment (niyobern) [reply vite.config.ts]: check now I've added it

File: public/vite.svg
Comment (dushimeemma) [inline public/vite.svg:1]: What is the use of this file?

File: src/assets/react.svg
Comment (dushimeemma) [inline src/assets/react.svg:1]: what is the use of this file?

**Review by dushimeemma — CHANGES_REQUESTED** [review]

File: src/assets/react.svg
Comment (niyobern) [reply src/assets/react.svg:1]: I simply pulled from develop
